pkg/comments/comment.go

```diff
@@ -0,0 +1,78 @@
+package comments
+
+import (
+	"strings"
+
+	"github.com/crossplane-contrib/terrajet/pkg/markers"
+)
+
+// Option is a comment option
+type Option func(*Comment)
+
+// WithReferenceTo returns a comment options with reference to input type
+func WithReferenceTo(t string) Option {
+	return func(c *Comment) {
+		c.ReferenceToType = t
+	}
+}
+
+// WithTFTag returns a comment options with input tf tag
+func WithTFTag(t string) Option {
+	return func(c *Comment) {
+		c.FieldTFTag = &t
+	}
+}
+
+// New returns a Comment by parsing Terrajet markers as Options
+func New(text string, opts ...Option) (*Comment, error) {
+	to := markers.TerrajetOptions{}
+
+	var lines []string
+	for _, line := range strings.Split(text, "\n") {
+		line = strings.TrimSpace(line)
+		if line == "" {
+			continue
+		}
+		// Only add raw marker line if not processed as an option (i.e. if it is
+		// not a Terrejet marker.) Terrajet markers will still be printed as
+		// comments while building from options.
+		parsed, err := markers.ParseAsTerrajetOption(&to, line)
+		if err != nil {
+			return nil, err
+		}
+		if !parsed {
+			lines = append(lines, line)
+		}
+	}
+
+	c := &Comment{
+		Text: strings.Join(lines, "\n"),
+		Options: markers.Options{
+			TerrajetOptions: to,
+		},
+	}
+
+	for _, o := range opts {
+		o(c)
+	}
+
+	return c, nil
+}
+
+// Comment represents a comment with text and supported marker options.
+type Comment struct {
+	Text string
+	markers.Options
+}
+
+// String returns a string representation of this Comment (no "// " prefix)
+func (c *Comment) String() string {
+	return c.Text + "\n" + c.Options.String()
+}
+
+// Build builds comments by adding comment prefix ("// ") to each line of
+// the string representation of this Comment.
+func (c *Comment) Build() string {
+	all := strings.ReplaceAll("// "+c.String(), "\n", "\n// ")
+	return strings.TrimSuffix(all, "// ")
+}
```

Comment:
```suggestion
// String builds comments by adding comment prefix ("// ") to each line of
// the string representation of this Comment.
func (c *Comment) String() string {
	all := strings.ReplaceAll("// "+c.String(), "\n", "\n// ")
	return strings.TrimSuffix(all, "// ")
}
```

nitpick: is there a case where the string representation of the comment is useful without `//`?

Reply:
Yes, it is used while building a comment with terrajet markers here: https://github.com/crossplane-contrib/provider-tf-aws/pull/10/commits/cfcdfa33d1e3bc2c0ac3b4d93b05abb4f2e8c394#diff-f59d142a3ad45f7a7bb182ca95d3d7130496d5269597d8217bb1ec16b63295cbR160

We parse that back in [here](https://github.com/crossplane-contrib/terrajet/pull/50/files#diff-f582aaab14f02ada1f64e64e80337f2a797b8e38ffe5c3288b38c98e67824a86R67). Always printing `//` would require us to be able to consider that while parsing.